### PR TITLE
Fixes: #10668 add local column to dependency rule error message

### LINKED ISSUE #10668: Add local column in dependency rule error message
### Describe the use case

I am currently migrating a rather large codebase to SQLAlchemy 2.0 and encountered the `Dependency rule tried to blank-out primary key...` error. I already know how to fix it in the codebase, but it took me like 20 minutes to fix it because the specified object has more than one affected relationship. I finally realised it when I printed out the local column before-hand.

I think it could be very handy if the error message contains the name of the local column.
My proposal would be: `Dependency rule on column '<local column>' tried to blank-out primary key column '<remote column>' on instance '<instance>'`

### Databases / Backends / Drivers targeted

*

### Example Use

The snippet from the codebase where the info about the local column would have helped (of course there are a lot more columns and methods involved in the real thing. I would have found my mistake immediately otherwise 😺 ):

```python

class Cable(DeclarativeBase):
    pnr: sa_orm.Mapped[int] = Column(Integer(), primary_key=True)
    id: sa_orm.Mapped[int] = Column(Integer(), primary_key=True)
    # error causing relationships come after

class Device(DeclarativeBase):
    pnr: sa_orm.Mapped[int] = Column(Integer(), primary_key=True)
    id: sa_orm.Mapped[int] = Column(Integer(), primary_key=True)
    # error causing relationships come after

class CadCableSide(DeclarativeBase):
    __tablename__ = "cad_cable_side"
   
    __table_args__ = (  # {{{
        ForeignKeyConstraint(["pnr"], ["project.pnr"], ondelete="CASCADE"),
        ForeignKeyConstraint(
            ["pnr", "device_id"],
            ["device.pnr", "device.id"],
            name="cad_cable_side_device_fk",
            ondelete="SET NULL",
        ),
        ForeignKeyConstraint(
            ["pnr", "cable_id"],
            ["cable.pnr", "cable.id"],
            name="cad_cable_side_cable_fk",
            ondelete="CASCADE",
        ),
    )

    pnr: sa_orm.Mapped[int] = Column(Integer(), primary_key=True)
    cable_id = Column(Integer, nullable=False)
    device_id = Column(Integer(), nullable=True)

    # error causing relationships come after

```

Deleting a cable or a device both result in the error because of the `pnr` column in both foreign keys. In my example I was trying to patch the CadCableSide->Cable relationship although I should have migrated CadCableSide->Device relationship instead. But the error message only said `Dependency rule tried to blank-out primary key column 'cad_cable_side.pnr' on instance '<CadCableSide ...>'`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jvllmr**
### Description

fixes #10668 
Added local column to dependency rule error message with the already present formatting method
### Checklist

This pull request is:

- [ ] A documentation / typographical / small typing error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision c7b62bf5e070950f05be83118cdd3f490ad37163 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change c7b62bf5e070950f05be83118cdd3f490ad37163: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4977

**Comment by zzzeek** [comment]
pretty good thanks for getting the tests

File: lib/sqlalchemy/orm/sync.py [hunk 1]
```diff
@@ -86,8 +86,10 @@ def clear(dest, dest_mapper, synchronize_pairs):
             not in orm_util._none_set
         ):
             raise AssertionError(
-                "Dependency rule tried to blank-out primary key "
-                "column '%s' on instance '%s'" % (r, orm_util.state_str(dest))
+                "Dependency rule on column '%s' "
+                "tried to blank-out primary key "
+                "column '%s' on instance '%s'"
+                % (l, r, orm_util.state_str(dest))
```
Comment (CaselIT) [inline lib/sqlalchemy/orm/sync.py:92]: Not required, but we could use f-strings here

**Review by sqla-tester — COMMENTED** [review]
Hi, this is **sqla-tester** and I see you've pinged me for review. However, user **jvllmr** is not authorized to initiate CI jobs.  Please wait for a project member to do this!

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision d7f9952b81394d585e55dad6d6f355aaa54f599e of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset d7f9952b81394d585e55dad6d6f355aaa54f599e added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4977

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

this needs a 2.0 backport

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4977

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

already done in https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4955?usp=cherry-pick

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4977

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4977 has been **merged**. Congratulations! :)

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4955 has been **merged**. Congratulations! :)

**Pull request closed without merging**